### ISSUE TITLE
Add Pipeline Trigger Extractor

### ISSUE THREAD

**Issue by Gasser278** [gitlab4j/gitlab4j-api#962]
If one looks at a pipeline using the pipeline API, there will be no information about the trigger source of that pipeline. If one wants to find out if the pipeline was triggered by a push, for example, this information would not be found. However, if one uses the HTTP request with the pipeline ID, this information will be found under "source". Therefore, it just needs to be added to the Java API, since the trigger provides important information about the pipeline.

**Comment by jmini** [comment]
I think your request is solved with https://github.com/gitlab4j/gitlab4j-api/pull/959

If not please elaborate on what is needed, maybe by adding some references to the java classes and/or some pointer to the gitlab documentation.

**Comment by Gasser278** [comment]
It would be solved with #959 but I can't use it because it hasn't been included in the latest release yet. Is there a way to add it to my dependencies right now or should I wait until the next release?

**Comment by jmini** [comment]
I hope it will be released soon.

To test in advance:

* you can build locally and use that version (I would change the version in the pom to `5.2.0.my-preview` instead of `5.2.0-SNAPSHOT`). But you often also need to publish it somewhere (Nexus Server, Artifactory Server, any HTTP server) so that it can be consumed from different places (like your CI, your team-mates, …)
* you can use [jitpack](https://jitpack.io/)

---

For Jitpack I have an example in Gradle when I was consuming a specific tag I created in my fork: [5.1.0-jmini.4](https://github.com/jmini/gitlab4j-api/releases/tag/5.1.0-jmini.4)

```gradle
repositories {
    mavenCentral()
    maven {
        url "https://jitpack.io"
        content {
            includeGroup "com.github.jmini"
        }
    }
}

dependencies {
    // ...
    implementation 'com.github.jmini:gitlab4j-api:5.1.0-jmini.4'
    // ...
}
```

This works also out of the box if you are running some script with [Jbang](https://www.jbang.dev/) by declaring the dependency like this:

```java
//DEPS https://github.com/jmini/gitlab4j-api/tree/5.1.0-jmini.4
```

Instead of:
```java
//DEPS org.gitlab4j:gitlab4j-api:5.1.0
```

And then Jbang does the work for you.
